### PR TITLE
Add timestamp to scan summary #925

### PR DESCRIPTION
**Pull request by techytushar**
PR for issue #925 
Signed-off-by: Tushar Mittal <chiragmittal.mittal@gmail.com>

### CODE REVIEW

File: src/scancode/cli.py [hunk 1]
```diff
@@ -47,7 +47,7 @@
 from scancode_config import scancode_temp_dir
 
 from commoncode.fileutils import PATH_TYPE
-from commoncode.timeutils import time2tstamp
+from commoncode.timeutils import time2tstamp,tstamp2time
```
Comment (pombredanne) [inline src/scancode/cli.py]: Please keep each import on a separate line, but also based on my other comments `tstamp2time` is not likely needed at all.

File: src/scancode/cli.py [hunk 2]
```diff
@@ -1202,7 +1202,14 @@ def display_summary(codebase, scan_names, processes, verbose):
                                 '%(final_size_count)s' % locals())
 
     echo_stderr('Timings:')
+
+    timestamp = codebase.timings.get('timestamp',None)
+    timestamp = tstamp2time(timestamp)
```
Comment (pombredanne) [inline src/scancode/cli.py]: Why do you get a time back?

File: src/scancode/cli.py [hunk 3]
```diff
@@ -1202,7 +1202,14 @@ def display_summary(codebase, scan_names, processes, verbose):
...
+    echo_stderr('  timestamp: %(timestamp)s'% locals())
+    
     for name, value, in codebase.timings.items():
+        if name == 'timestamp':
```
Comment (pombredanne) [inline src/scancode/cli.py]: You should pop() above instead IMHO
e.g. not `timestamp = codebase.timings.get('timestamp',None)` but instead `timestamp = codebase.timings.pop('timestamp',None)`

File: src/scancode/cli.py [hunk 4]
```diff
@@ -724,7 +724,7 @@ def scancode(ctx, input,  # NOQA
         # TODO: this is weird: may be the timings should NOt be stored on the
         # codebase, since they exist in abstract of it??
         codebase.timings.update(setup_timings)
-
+        codebase.timings['timestamp'] = scan_start
```
Comment (pombredanne) [inline src/scancode/cli.py]: IMHO the scan_start should be its own attribute on the Codebase. This is not a "timing" proper.

File: src/scancode/cli.py (hunk 1 above)
Comment (techytushar) [reply src/scancode/cli.py]: IMHO `time2tstamp` returns date time like this `2018-02-19T145640.979020` which is quite hard to read and understand so i converted it to `2018-02-19 14:56:40.979020` using `tstamp2time`

Comment (pombredanne) [reply src/scancode/cli.py]: IMHO It is pure luck if a datetime ends up being serialized alright.
Instead you should use this I just pushed https://github.com/nexB/scancode-toolkit/commit/179d983a222cb067a40cfdcf05de70da1ffa262f with `path_safe=False`

File: src/scancode/cli.py (hunk 3 above)
Comment (pombredanne) [reply src/scancode/cli.py]: But this should not be in timings... but a proper attribute, so this is moot.

**Comment by techytushar** [comment]
@pombredanne please review, i have made the required changes.. :sweat_smile:

File: src/formattedcode/output_json.py [hunk 5]
```diff
@@ -92,18 +93,21 @@ def process_codebase(self, codebase, output_json_pp, files_count,
...
                    scancode_notice=scancode_notice,
+                   timestamp = codebase.scan_start,
                    pretty_options=pretty_options,
                    pretty=True)
 
 
 def write_json(results, output_file, files_count,
                scancode_version, scancode_notice,
-               pretty_options, pretty=False):
+               timestamp,pretty_options,
```
Comment (pombredanne) [inline src/formattedcode/output_json.py]: your code is not properly formatted

**Comment by techytushar** [comment]
@pombredanne please review, add the test

File: src/scancode/cli.py [hunk 6]
```diff
@@ -1202,6 +1202,10 @@ def display_summary(codebase, scan_names, processes, verbose):
                                 '%(final_size_count)s' % locals())
 
     echo_stderr('Timings:')
+
+    timestamp = codebase.scan_start
+    echo_stderr('  timestamp: %(timestamp)s'% locals())
```
Comment (pombredanne) [inline src/scancode/cli.py]: Could we call this Scan start instead here and in the JSON output? ... since this is what this is.

**Comment by techytushar** [comment]
@pombredanne only one test is failing, but all tests pass on my local machine, please review..

**Comment by pombredanne** [comment]
The travis failure is an heisenbug. I respun a build

**Comment by pombredanne** [comment]
Merging now. Thanks!

**Pull request merged**